infrasec/tutorials/atlantis.md

```diff
@@ -0,0 +1,418 @@
+
+3. Validate the Route53 DNS records in the Commercial account using the [`aws_acm_certificate_validation` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate_validation).
+
+    We'll need to find the `arn` to plug into the [atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) we'll call in the next step. We can find this in the console's Certificate Manager after merging the PR to add the certificate.
+
+    <img src="https://github.com/trussworks/Engineering-Playbook/blob/3efe6ea02ed010f3db2c07921c5c8acc60406b84/infrasec/tutorials/images/atlantis_acm1.png" width="450">
+
+### Set up Docker 🐋
+
+For any project not requiring a specific type of docker implementation, we can simply [pass in the `trussworks-atlantis-ecs-image` image](https://github.com/trussworks/trussworks-atlantis-ecs-image) when we make the module call in [Step 3](#step-3-call-the-atlantis-module-).
+
+If we do not set the `atlantis_image` variable, we'll find `atlantis:latest` is used by default. This default is not recognized as updated when a new "latest" is released due to the word remaining unchanged. Therefore, we recommend passing in a numbered version of the Atlantis docker image to the `atlantis_image` var.
+
+Note that the use of Docker in tandem with Atlantis is actually optional, however both our example usage here and the Atlantis Fargate module expect it.
+
+## Step 3: Call the [Atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) 🧜
+
+We'll start with the most basic Atlantis module call we can. Assuming an existing VPC, Robot user, validated certificate, zone name, Atlantis Docker image, and GitHub connection, the code will look something like this
+
+  ```hcl
+  module "atlantis" {
+    source  = "terraform-aws-modules/atlantis/aws"
+    version = "~> 2.36.0" // make sure to update to the latest version
+
+    name = "${var.environment}-atlantis"
+
+    atlantis_image = "milmove/circleci-docker:milmove-atlantis"
+
+    # Route53
+    create_route53_record = false
+    route53_zone_name = "atlantis.exp.move.mil"
+    certificate_arn = "arn:aws-us-gov:acm:us-gov-west-1:015681133840:certificate/456c11bd-a730-476d-ba7c-1bd4690ede3f"
+
+    # VPC
+    vpc_id          = ""
+    cidr            = "10.30.0.0/16"
+    azs             = ["us-gov-east-1a", "us-gov-east-1b", "us-gov-east-1c"]
+    private_subnet_ids = ["subnet-012345678f01", "subnet-012345678f02", "subnet-012345678f03"]
+    public_subnet_ids  = ["subnet-012345678f04", "subnet-012345678f05", "subnet-012345678f06"]
+
+    # Atlantis
+    atlantis_github_user = "atlantis"
+    atlantis_github_user_token = data.aws_ssm_parameter.github_user_token.value
+    atlantis_repo_whitelist          = ["github.com/transcom/transcom-infrasec-gov"]
+    atlantis_hide_prev_plan_comments = "true"
+    allow_github_webhooks = true
+  }
+  ```
```

Comment:
Do not use real AWS account numbers, github repos, etc here. In addition, I'm curious why you're specifying the certificate arn directly rather than a resource (eg `aws_acm_certificate.my_cert.arn`).

Reply:
yeah I wanted to make the value expectations explicit to reflect a first pass. Fwiw this was my final code. 

```hcl
module "atlantis" {
  source  = "terraform-aws-modules/atlantis/aws"
  version = "~> 2.39.0"

  name = "atlantis-${var.environment}"

  # Image managed by repo https://github.com/XXX
  atlantis_image = "milmove/circleci-docker:XXX"

  # Route53
  create_route53_record = false
  route53_zone_name     = var.zone_name
  certificate_arn       = var.certificate_arn

  # VPC
  vpc_id             = var.vpc_id
  cidr               = var.vpc_cidr
  azs                = var.vpc_azs
  private_subnet_ids = var.vpc_private_subnet_ids
  public_subnet_ids  = var.vpc_public_subnet_ids

  # Atlantis
  atlantis_github_user       = "robot-mymove"
  atlantis_repo_whitelist    = ["myRepo"]
  # The module requires this to be a string
  # despite looking like a bool 🤷‍♀️
  atlantis_hide_prev_plan_comments = "true"
  allow_github_webhooks            = true

  # These policy arns are passed through to the submodule
  # which creates the ecs instance role and permissions
  policies_arn       = var.ecs_task_policies_arn
  trusted_principals = var.ecs_trusted_principals

  # We use custom environment secrets rather than the module inputs
  # because the module is buggy
  # see https://trussworks.slack.com/archives/CLNC1MUBS/p1615847664002200?thread_ts=1615570522.023200&cid=CLNC1MUBS
  custom_environment_secrets = [
    {
      name      = "SSM_ATLANTIS_GH_TOKEN"
      valueFrom = "/atlantis/github/user/token"
    },
    {
      name      = "SSM_ATLANTIS_GH_WEBHOOK_SECRET"
      valueFrom = "/atlantis/webhook/secret"
    },
  ]

  custom_environment_variables = [
    {
      name  = "ATLANTIS_REPO_CONFIG_JSON"
      value = file("atlantis.json")
    },
    { "name" : "ATLANTIS_WRITE_GIT_CREDS",
      "value" : "1"
    },
  ]

  # ALB
  alb_log_bucket_name     = var.alb_log_bucket_name
  alb_log_location_prefix = var.alb_log_location_prefix
  alb_logging_enabled     = true
  alb_ingress_cidr_blocks = []
}
```